src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -0,0 +1,507 @@
+                    int verifyTimeout, int appUpdateTimeout, double compileWait, boolean libertyDebug
+        ) throws IOException {
+            super(serverDirectory, sourceDirectory, testSourceDirectory, configDirectory, resourceDirs,
+                    hotTests, skipTests, false, false, artifactId,  serverStartTimeout,
+                    verifyTimeout, appUpdateTimeout, ((long) (compileWait * 1000L)), libertyDebug);
+
+            ServerFeature servUtil = getServerFeatureUtil();
+            this.existingFeatures = servUtil.getServerFeatures(serverDirectory);
+        }
+
+        @Override
+        public void debug(String msg) {
+            logger.debug(msg);
+        }
+
+        @Override
+        public void debug(String msg, Throwable e) {
+            logger.debug(msg, e);
+        }
+
+        @Override
+        public void debug(Throwable e) {
+            logger.debug(e);
+        }
+
+        @Override
+        public void warn(String msg) {
+            logger.warn(msg);
+        }
+
+        @Override
+        public void info(String msg) {
+            logger.info(msg);
+        }
+
+        @Override
+        public void error(String msg) {
+            logger.error(msg);
+        }
+
+        @Override
+        public void error(String msg, Throwable e) {
+            logger.error(msg, e);
+        }
+
+        @Override
+        public boolean isDebugEnabled() {
+            return logger.isEnabled(LogLevel.DEBUG);
+        }
+
+        @Override
+        public void stopServer() {
+            if (isLibertyInstalled(project)) {
+                if (getServerDir(project).exists()) {
+                    ServerTask serverTaskStop = createServerTask(project, "stop");
+                    serverTaskStop.execute();
+                } else {
+                    logger.error('There is no server to stop. The server has not been created.');
+                }
+            } else {
+                logger.error('There is no server to stop. The runtime has not been installed.');
+            }
+        }
+
+        @Override
+        public ServerTask getServerTask() throws Exception {
+            if (serverTask != null) {
+                return serverTask;
+            }
+
+            copyConfigFiles();
+
+            if (libertyDebug) {
+                serverTask = createServerTask(project, "debug");
+                setLibertyDebugPort(libertyDebugPort);
+                serverTask.setEnvironmentVariables(getDebugEnvironmentVariables());
+            } else {
+                serverTask = createServerTask(project, "run");
+            }
+
+            serverTask.setClean(clean);
+
+            return serverTask;
+        }
+
+        @Override
+        public List<String> getArtifacts() {
+            // https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management
+            String[] dependencyConfigurationNames = ['compile', 'compileOnly', 'testCompileOnly'];
```

Comment:
Users could also be using the `api` or `implementation` [configurations](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation). Since you can't access `implementation` directly you should pull the dependency list from `compileClasspath` with `project.configurations.compileClasspath.resolvedConfiguration.resolvedArtifacts.each`. That should cover all the different compile level configurations.

Reply:
@mattbsox According to that document, `implementation` inherits `api` so it may be enough to just add `api` to this list.

Reply:
Okay, sounds good to me. Try using both of those configurations somewhere in the test cases if possible.

I need to check if we need to update any other parts of the plugin to account for these.

Reply:
Changed to use `compileClasspath` which extends `compile, compileOnly, implementation` and `testCompileClasspath` which extends `testCompile, testCompileOnly, testImplementation`